pyne/ensdf_processing.py

```diff
@@ -5,40 +5,82 @@
 from pyne.utils import QAWarning
 
 try:
-    import urllib.request as urllib2
+    import urllib.request as urllib
 except ImportError:
-    import urllib2
+    import urllib2 as urllib
 
 if sys.version_info[0] > 2:
     basestring = str
 
 warn(__name__ + " is not yet QA compliant.", QAWarning)
 
+class SetupIncompleteError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+            return "This ensdf analysis module requires additional setup.  Run \
+                    pyne.ensdf_processing.setup_additional_downloads to download the \
+                    additional executables and complete this modules instalation. \
+                    (following executable not yet installed/downloaded: " + repr(self.value)
+
 def path_to_exe(exe_name):
     exe_path_abs, dp = os.path.split(os.path.abspath(__file__))
     exe_path_abs = os.path.join(exe_path_abs, exe_name)
     exe_path_abs = os.path.join('./',exe_path_abs)
     return exe_path_abs
 
-def verify_download_exe(exe_path, exe_url, compressed = 0, decomp_path = '', dl_size = 0):
+def download_exe(exe_path, exe_url, compressed = 0, decomp_path = '', dl_size = 0):
     if not os.path.exists(exe_path):
-        print('fetching executable')
-        response = urllib2.urlopen(exe_url)
-        prog = 0
-        CHUNK = 32 * 1024
-        f = open(exe_path, 'wb')
-        while True:
-            chunk = response.read(CHUNK)
-            prog = prog + (256)
-            if not chunk: break
-            f.write(chunk)
-        f.close()
+        msg = 'Downloading {0!r} to {1!r}'.format(exe_url, exe_path)
+        print(msg)
+        req = urllib.Request(exe_url, headers={'User-Agent': 'Mozilla/5.0'})
+        f = urllib.urlopen(req, timeout=30.0)
+        try:
+            html = f.read()
+        finally:
+            f.close()
+        with open(exe_path, 'wb') as f:
+            f.write(html)
+
         # set proper permissions on newly downloaded file
         os.chmod(exe_path, 744)
         if compressed:
+            os.chmod(exe_path, 777)
             tfile = tarfile.open(exe_path, 'r:gz')
             tfile.extractall(decomp_path)
 
+def setup_additional_downloads():
+    print("Downloading BRICC executable")
+    setup_bricc()
+    print("Downloading GABS executable")
+    setup_gabs()
+    print("Downloading RADLIST executable")
+    setup_radlist()
+
+def setup_bricc():
+    exe_path = path_to_exe('bricc')
+    exe_dir = path_to_exe('')[:-1]
+    compressed_exe_path = exe_path + '.tar.gz'
+
+    bricc_url = "http://www.nndc.bnl.gov/nndcscr/ensdf_pgm/analysis/BrIcc/Linux/BriccV23-Linux.tgz"
```

Comment:
This line is over the 80-char limit, but I'm not sure how to handle that stylistically given that the URL itself is 81 chars... @scopatz ?